framework/src/mediaplayer/mediaplayer.c

```diff
@@ -0,0 +1,530 @@
+#include <tinyara/config.h>
+#include <tinyara/audio/audio.h>
+#include <tinyara/audio/pcm.h>
+#include <tinyalsa/tinyalsa.h>
+#include <tinyara/fs/ioctl.h>
+#include <pthread.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sched.h>
+
+#include <mediaplayer/mediaplayer.h>
+
+/***
+ * Structure
+ */
+struct media_s {
+	dq_entry_t q;
+
+	int fd;
+	media_op_e op;
+	media_state_e state;
+	media_type_e type;
+};
+
+/***
+ * Static variables
+ */
+static pthread_t g_pth_playing;
+#ifdef CONFIG_AUDIO_MULTI_CARD
+static pthread_t g_pth_recording;
+#endif
+static sem_t g_sem_playing;
+static sem_t g_sem_recording;
+static pthread_mutex_t g_mutex_playing_q;
+static pthread_mutex_t g_mutex_recording_q;
+
+static bool g_playing_live;
+static bool g_recording_live;
+
+static dq_queue_t g_playing_q;
+static dq_queue_t g_recording_q;
+
+/***
+ * Pre-definitions
+ */
+int threadfunc_audio_playing(void *args);
+int threadfunc_audio_recording(void *args);
+
+int media_init(struct media_cb_s *cbs)
+{
+	pthread_attr_t attr;
+	pthread_mutexattr_t mutexAttr;
+	struct sched_param sparam;
+
+	// Init dq and sem
+	dq_init(&g_playing_q);
+	sem_init(&g_sem_playing, 0, 0);
+
+	pthread_mutexattr_init(&mutexAttr);
+	pthread_mutexattr_settype(&mutexAttr, PTHREAD_MUTEX_NORMAL);
+	pthread_mutex_init(&g_mutex_playing_q, &mutexAttr);
+
+
+#ifdef CONFIG_AUDIO_MULTI_CARD
+	dq_init(&g_recording_q);
+	sem_init(&g_sem_recording, 0, 0);
+	pthread_mutex_init(&g_mutex_recording_q, &mutexAttr);
+#endif
+	// Init threads
+	if (pthread_attr_init(&attr) != 0) {
+		return -MEDIA_ERROR_THREAD_INIT;
+	}
+
+	sparam.sched_priority = 150;
```

Comment:
@btheosam Benix could you please let us know your opinion about this? configurable value needed? or keep hard coded value? Also please check thread modeling here.